### ISSUE TITLE
[Feature request] Add more low-level MSExperiment multidimensional access functions

### ISSUE THREAD

**Issue by jpfeuffer** [OpenMS/OpenMS#7248]
1) IMHO e.g. functions for trace/chrom. extraction are completely scattered around and have 100 different names and are often only applicable to very specific use cases
2) Would make the library much more useful in pyopenms since you would not need to go in and out of python all the time

Just something like:
exp.extract(rt_min, rt_max, mz_min, mz_max, im_min, im_max, ms_levels, aggregate_rt, aggregate_mz, aggregate_im)

Or some kind of:
exp.extract(rt_min, rt_max, mz_min, mz_max, im_min, im_max, ms_levels,).aggregate(axis, function)
(where extract returns a view or something)

Where aggregate is an optional function (or at least a set of predefined ones like min, max, mean, sum, median, count).

**Comment by timosachsenberg** [comment]
I get your point, but wonder how we can do this consistently.
Using modern c++ the most natural one would probably be returning a view (as you said) that behaves similar to a pair of AreaIterators.
But would the API really be that simple? E.g., one often wants to preserve, e.g., the min peak per RT range while summing/integrating over RT? Can this be easily modelled?

**Comment by jpfeuffer** [comment]
Yes, happy to hear your input, otherwise I just add yet another iteration scheme just for pyopenms with a fixed aggregation function for my needs.

Maybe we can gather common access patterns here? I did not completely understand yours. Minimum in RT but integrate over RT?

One problem that I already saw is the not-so-native IM support. You need to sort in specific ways for "fast" access in specific dimensions. But I'm happy to ignore/accept slow IM access.

**Comment by timosachsenberg** [comment]
Sorry I meant max peak. Eg for chromatogram extraction one often want to take only the highest peak in the m/z tolerance window per spectrum. And then sum up those highest peaks. Or calculate an area etc

**Comment by timosachsenberg** [comment]
if you get something like: exp.extract(...).aggregate(AXIS::MZ, maxIntensity).aggregate(AXIS::RT, simpson); to work it would be great
because that is pretty much what a range interface could do
exp | extract(...) | aggregate(AXIS::MZ, maxIntensity) | aggregate(AXIS::RT, simpson) | toXXX;

if too much work, maybe easier to implement (also to wrap in python) something like
exp.extract(...).aggregate(maxIntensity(), simpson(), mean())
and assume a fixed order of evaluation

**Comment by jpfeuffer** [comment]
Yes I think Ill go for your last suggestion 
a) because of pyopenms
b) we probably actually want lazy evaluation. We do not want an intermediate datastructure to offer a new range. (we would probably need coroutines for this to generatively yield)

**Comment by timosachsenberg** [comment]
I think we implemented that
